test/lib/clients/inst_productsources_test.rb

```diff
@@ -46,6 +49,12 @@
 
       client.main
     end
+
+    it "sets the ZYPP_REPO_RELEASEVER environment variable to Product.version" do
+      expect(ENV).to receive(:[]=).with("ZYPP_REPO_RELEASEVER", "15.1")
```

Comment:
well, issue is that also other tests modify that env.
so I suggest to add something like
```
around do |test|
      old_env = ENV.to_hash
      test.call
      ENV.replace(old_env)
end
```

I use e.g. also in cheetah https://github.com/openSUSE/cheetah/blob/master/lib/cheetah.rb#L416

Reply:
BTW you no longer need to mock this and can simply check ENV as it was before.